lambda_function.py

```diff
@@ -0,0 +1,93 @@
+import json
+import os
+import logging
+from google.cloud import bigquery
+
+
+def lambda_handler(event):
+    """
+    Parses messages sent to liveclassAttendanceEventHandler lambda function..
+    Each message needs to have the following fields for row to be inserted:
+        - dateTime
+        - purpose
+            - type
+            - subType
+            - params
+                - platform
+                - id
+        - authType
+        - user
+            - values
+            - userDataValidated
+    """
+    messageBody = event["Records"]
+
+    for eachMessage in messageBody:
+        if "Sns" in eachMessage and "Message" in eachMessage["Sns"]:
+            # extracting the body of the message
+            message = json.loads(eachMessage["Sns"]["Message"])[0]
+
+            row = {}
+
+            # check if the key values exist in the message sent
+            # if all the fields don't exist,
+            #   the row isn't inserted and an error is displayed
+
+            if all(
+                (k in message for k in ("dateTime", "purpose", "authType", "user"))
+                and (k in message["purpose"] for k in ("type", "subType", "params"))
+                and (k in message["user"] for k in ("values", "userDataValidated"))
```

Comment:
This looks like it's looking only for `user-auth` related functionality, no?

Doesn't the same lambda handle the attendance SQS messages also?

Reply:
I didn't get your question. This lambda takes in all attendance related messages. Checks if all the fields are present and adds the row to BQ.

Reply:
Ah, my bad. I got confused and thought this was handling all SQS messages 👍🏾